### PR TITLE
Emit lifecycle log in the destroy method when pipeline is run in preview mode.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-9326

When run in preview mode, we do not execute the post actions. We were retuning from the SmartWorkflow.destroy method, which causes log statement not to show up.

### CODE REVIEW

**Comment by yaojiefeng** [comment]
LGTM

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java [hunk 1]
```diff
@@ -197,22 +197,23 @@ public void initialize(WorkflowContext context) throws Exception {
...
+      for (Map.Entry<String, PostAction> endingActionEntry : postActions.entrySet()) {
+        String name = endingActionEntry.getKey();
+        PostAction action = endingActionEntry.getValue();
+        StageInfo stageInfo = StageInfo.builder(name, PostAction.PLUGIN_TYPE).build();
+        BatchActionContext context = new WorkflowBackedActionContext(workflowContext, workflowMetrics, lookupProvider,
+                                                                     logicalStartTime, runtimeArgs, stageInfo);
+        try {
+          action.run(context);
+        } catch (Throwable t) {
+          LOG.error("Error while running ending action {}.", name, t);
```
Comment (nitinmotgi) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java]: What's a end action ?

Comment (sagarkapare) [reply cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java]: Its actually post action.  Updated log message to 

```
LOG.error("Error while running post action {}.", name, t);
```

**Comment by sagarkapare** [comment]
Build is running here - https://builds.cask.co/browse/CDAP-RUT888-1

**Comment by sagarkapare** [comment]
Build is green. Merging.

**Pull request merged**